### ISSUE TITLE
Improve session handling for better compatibility with various hosts.

### ISSUE THREAD

**Issue by pderksen** [moonstonemedia/WP-Simple-Pay-Lite-for-Stripe#38]
_TODO: Bring more details & findings in from issue in Pro repo._

Hoping this might be helpful to work out between EDD, Give, Ninja Forms & WP Simple Pay. Possibly others I don't know about. Reposting this in existing related issues on each repo to discuss.

- EDD: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/5418
- Give: https://github.com/WordImpress/Give/issues/1630
- Ninja Forms: https://github.com/wpninjas/ninja-forms/issues/3175
- WP Simple Pay Lite: https://github.com/moonstonemedia/WP-Simple-Pay-Lite-for-Stripe/issues/38

Up until now we've been using the core of [WP Session Manager](https://github.com/ericmann/wp-session-manager) plugin to manage sessions within WP Simple Pay ([Lite](https://github.com/moonstonemedia/WP-Simple-Pay-Lite-for-Stripe) & Pro).

However, more recently this hasn't been working on some hosts such as Flywheel & Patheon. After doing some digging & testing, we're now beta testing the core of Patheon's [WP Native PHP Sessions](https://github.com/pantheon-systems/wp-native-php-sessions) plugin. So far beta tests on various hosts are working out great.

Any reason not to go this direction and include WP Native PHP Sessions like this? Is there an easier more straightforward way to handle sessions across modern managed hosting in our plugins?

**Comment by pderksen** [comment]
Went back to WP Session Manager due to conflicts with other plugins calling `session_start()`, but now using the current version (2.0.2) that by default uses a custom table to get/set session values.

Seems to be working so far _except_ for on Flywheel & Pantheon in my testing. See https://github.com/ericmann/wp-session-manager/issues

**Comment by pderksen** [comment]
We're back to using WP Session Manager within WP Simple Pay, except we're now using the code from the latest version of it (2.0.2).

We ran into issues using WP Native PHP Sessions by Pantheon when other plugins using `session_start()` were active.

This seems to work on all hosts we tested _except_ Pantheon.

So now we default to the use of WP Session Manager's table data storage method, but we also have a constant that allows the site owner to force use of PHP sessions.

Finally, we check for the existence of WP Native PHP Sessions (this plugin) and force use of PHP sessions if detected.
